src/CTA.WebForms2Blazor/FileConverters/CodeFileConverter.cs

```diff
@@ -58,19 +50,44 @@ public class CodeFileConverter : FileConverter
             }
         }
 
-
         public override async Task<IEnumerable<FileInformation>> MigrateFileAsync()
         {
             LogStart();
 
-            var classMigrationTasks = _classConverters.Select(classConverter => classConverter.MigrateClassAsync());
+            // Store migration tasks alongside converter so we can log context where necessary
+            var classMigrationTasks = new List<(ClassConverter, Task<IEnumerable<FileInformation>>)>();
+
+            foreach (var converter in _classConverters)
+            {
+                classMigrationTasks.Add((converter, converter.MigrateClassAsync()));
+            }
 
             // We want to do our cleanup now because from this point on all migration tasks
             // are done by class converters and we want to make sure that we retire the task
             // related to this file converter before we await
             DoCleanUp();
 
-            var result = (await Task.WhenAll(classMigrationTasks)).SelectMany(newFileInformation => newFileInformation);
+            var allMigrationTasks = Task.WhenAll(classMigrationTasks.Select(t => t.Item2));
+
+            try
+            {
+                await allMigrationTasks;
+            }
+            // NOTE: We don't provide an reference to the caught exception because it would just be a
+            // standard aggregate exception of the exceptions that arose in the migration tasks, instead
+            // we choose to log an exception for each of the failed tasks to provide more info
+            catch
```

Comment:
shouldn't we catch    catch (AggregateException ex){}

Reply:
I don't think we would benefit from the reference to the aggregate exception as in the catch block we are enumerating then logging the exceptions that would make up the aggregate anyway. I think this is preferable to logging the aggregate exception because it will make our metrics on number and type of exceptions more accurate